docs/blog/_posts/2019-03-04-13th-dotty-milestone-release.md

```diff
@@ -0,0 +1,439 @@
+merely an alignment of IFTs into the new scheme.
+
+You can read about the alternative to implicits through the *Contextual
+Abstractions* section of our documentation or for a deep dive from the relevant
+PR chain that originated from
+[#5458](https://github.com/lampepfl/dotty/pull/5458). The syntax changes for new
+implicits are summarized in
+[#5825](https://github.com/lampepfl/dotty/pull/5825).
+
+This release offers the support for _typeclass derivation_ as a language
+feature. Typeclass derivation is a way to generate instances of certain type
+classes automatically or with minimal code hints, and is now supported natively
+with *dedicated language support*. A type class in this sense is any trait or
+class with a type parameter that describes the type being operated on. Commonly
+used examples are `Ordering`, `Show`, or `Pickling`. We introduce a new
+`derives` clause that generates implied instances of the `Eql`, `Ordering`, and
+`Pickling` traits in the companion object `Tree`. Take a look at the example
+below:
+
+```scala
+enum Tree[T] derives Eql, Ordering, Pickling {
+  case Branch(left: Tree[T], right: Tree[T])
+  case Leaf(elem: T)
+}
+```
+
+where the generated implied instances are the ones below:
+```scala
+implied [T: Eq]       for Eq[Tree[T]]       = Eq.derived
+implied [T: Ordering] for Ordering[Tree[T]] = Ordering.derived
+implied [T: Pickling] for Pickling[Tree[T]] = Pickling.derived
+```
+
+Note, the new syntax:
+
+```scala
+A extends B, C { ... }
+```
+
+which replaces:
+
+```scala
+A extends B with C { ... }
+```
+
+With typeclass derivation we can also derive types. A trait or class can appear
+in a derives clause if its companion object defines a method named `derived`.
+The type and implementation of a `derived` method are arbitrary, but typically
+it has a definition like this:
+
+```scala
+def derived[T] given Generic[T] = ...
+```
+
+You can read more about [Typeclass
+Derivation](https://dotty.epfl.ch/docs/reference/contextual/derivation.html) or
+for a deep dive at the relevant PRs:
+[#5540](https://github.com/lampepfl/dotty/pull/5540) and
+[#5839](https://github.com/lampepfl/dotty/pull/5839).
+
+_Multiversal equality_ is now supported through the `Eql` marker trait (renamed
+from `Eq` to differentiate it from Cats' `Eq`). For example, in order to be able
+to compare integers with strings now, instead of a custom implicit we can
+provide a derived implicit instance:
+
+```scala
+implied for Eql[Int, String] = Eql.derived
+```
+
+You can read how we based multiversal equality on typeclass derivation through
+the relevant PR [#5843](https://github.com/lampepfl/dotty/pull/5843).
+
+_Implicit conversions_ are now defined by implied instances of the
+`scala.Conversion` class. For example:
+
+```scala
+implied for Conversion[String, Token] {
+  def apply(str: String): Token = new KeyWord(str)
+}
+```
+
+_Top level_ definitions are now supported. This means that package objects are

```

Comment:
This should be a separate section with its own heading `## Introducing top level definitions`, and I'd move it above the long description of implicits.

Reply:
I will put on on a separate heading but I'll leave it after implied (ordered-by importance 🥇)

Reply:
> ordered-by importance

That's reasonable, my reasoning was that we should put the short and easy-to-understand stuff first to make sure people actually read it, otherwise I'm afraid the implied stuff will eclipse everything else. One way to avoid that would be to move the long description of the implied stuff in the documentation and link it from the blog post.

Reply:
I agree with @smarter. Put the short stuff first. Also it's a non-controversial, super-positive change that will bring some feel-good sentiment.

Reply:
The implied section is just a medley of all the various elements from the documentation so it will not add any value to it. The remaining two (implicit resolution changes and SemanticDB generator) cannot be on top of implied so I propose lets keep this order. However, I improved the very first sentence to reflect the updated title 😉

Reply:
Oh well, you won! 😆I am moving the top level after spark.

Reply:
The 

```scala
implied {
  def (x: C) pair (y: C) = (x, y)
}
```

has to go now, as it is a spoiler alert for the section to come.

Reply:
It's also syntax that isn't introduced.  (What even is it?  An implied block of definitions?  Can't get it to do anything useful in the REPL)

Reply:
https://dotty.epfl.ch/docs/reference/contextual/relationship-implicits.html#anonymous-implied-instances